tests/saw/spec/handshake/handshake.saw

```diff
@@ -24,6 +24,10 @@
 //
 ////////////////////////////////////////////////////////////////////////////
 
+// NB: handshake_io_lowlevel.saw enables lax_loads_and_stores, which is
```

Comment:
I had to look up what "NB" means in a comment. I'd probably make the warning more obviously scary.

What are the consequences of this? Like what tradeoffs are we accepting?

Reply:
> I had to look up what "NB" means in a comment. I'd probably make the warning more obviously scary.

Fair enough. I've replaced this with "WARNING" in an attempt to make it scarier.

> What are the consequences of this? Like what tradeoffs are we accepting?

Aside from only being able to use What4-based tactics, the other tradeoff from using `enable_lax_loads_and_stores` that comes to mind is that SAW will be more forgiving when it reads from uninitialized memory. For example, supposing you had this C code:

```c
struct s {
  int8_t x;
  int8_t y;
};

int8_t get_y(struct s *ss) {
  return ss->y;
}
```

If you wrote a SAW specification that forgot to explicitly set the `y` field in the `struct s` arguments, e.g.,

```
enable_experimental;
enable_lax_loads_and_stores;

let get_y_spec = do {
  ss <- llvm_alloc (llvm_alias "struct.s");
  // I forgot to use `llvm_points_to (llvm_field ss "y") ...` here
  llvm_execute_func [ss];
};
```

This specification will fail without `enable_lax_loads_and_stores`, as simulating the C code will fail unless the `y` field is explicitly initialized. It will succeed with `enable_lax_loads_and_stores`, however, as it will simply return a fresh, symbolic value and move on. It's like if every use of `llvm_alloc` is implicitly followed by symbolic invocations of `llvm_points_to` of the appropriate shape.

The downside to the `enable_lax_loads_and_stores` approach is that the C code simulator will proceed even if the specification doesn't explicitly specify the initial values of all arguments involved. This means that some extra care is warranted to make sure all of your preconditions are stated explicitly in your specifications, because if you leave them out, SAW will fill them in for you with very underconstrained assumptions. It shouldn't cause any unsoundness, however (where soundness is defined modulo uninitialized inputs).

-----

I've added an abridged version of this to the comments in `handshake_io_lowlevel.saw`.